OSDOCS-20108:Update the z-stream RNs for 4.21.20 (#113191)
openshift/openshift-docs: modules/zstream-4-21-20.adoc

* Before this update, when every service step could run entirely out-of-band (that is, there was no agent ramdisk), the conductor had already skipped the initial ramdisk boot. However, an internal flag still indicated that a ramdisk boot was allowed. As a consequence, steps that rebooted the node to finish their work could still trigger an agent ramdisk boot before that reboot, even though none of the steps required it. This issue caused Operators to identify an extra, unnecessary agent ramdisk boot during out-of-band service work that was disruptive and slower, such as when applying BIOS settings over the management interface. With this release, whenever all service steps are out-of-band-only, the conductor now sets that internal flag so the `no ramdisk needed` decision also applies consistently through the reboot-to-finish paths. As a result, the service-step flows do not pull in a spare agent boot. Instead, the node stays on the intended out-of-band path without the extra cycle.  (link:https://redhat.atlassian.net/browse/OCPBUGS-82294[OCPBUGS-84370])

> * Before this update, when editing secrets through the Web Console, binary data such as `JAR` or `JCEKS` keystores, was incorrectly decoded as UTF-8 text, which corrupted the binary values with replacement characters and rendered the files unusable. With this release, the Console now preserves base64-encoded binary data and passes it directly to the file input component without text conversion. Binary secret data remains intact when editing other fields in the same secret. (link:https://issues.redhat.com/browse/OCPBUGS-85674[OCPBUGS-85674])

Reviewer: ```suggestion
* Before this update, binary data such as `JAR` or `JCEKS` keystores, was incorrectly decoded as UTF-8 text when `Secret` objects were edited through the {ocp} web console. This corrupted the binary values with replacement characters and rendered the files unusable. With this release, the web console now preserves base64-encoded binary data and passes it directly to the file input component without text conversion. Binary secret data remains intact when editing other fields in the same secret. (link:https://issues.redhat.com/browse/OCPBUGS-85674[OCPBUGS-85674])
```
Please read and edit all of these highly verbose Renoa outputs, instead of leaving it to the merge reviewers: I know I have made this style comment about the web console numerous times

* Before this update, a race condition occurred between network namespace management and the optional container mount namespace separation in the rare instance when software on the host that was not within the {product-title} cluster created network namespaces. As a consequence, pods became stuck and could not restart. With this release, the race condition is solved by initializing the network namespace infrastructure before the container mount namespace. (link:https://issues.redhat.com/browse/OCPBUGS-86003[OCPBUGS-86003])

* Before this update, the `MachineSet` object scale sub-resource lacked a valid selector, which prevented autoscalers like HPA and KEDA from scaling the `MachineSet` object. As a consequence, autoscalers requiring a selector failed. With this release, the `MachineSet` object exposes an active label selector on the scale sub-resource, which enables scaling using autoscalers like HPA and KEDA that require the label selector to be populated. (link:https://issues.redhat.com/browse/OCPBUGS-86493[OCPBUGS-86493])

> * Before this update, when a user applied a `MachineConfig` parameter to install extensions, the Machine Config Operator (MCO) did not validate that all packages were installed. As a consequence, extension installation appeared to be successful, but packages were missing. With this release, the MCO validates after the node reboot that all packages associated with your desired extension were successfully installed before reporting a successful update. As a result, if one of more package is not present, the node, and subsequently the associated MachineConfigPool, will degrade. (link:https://issues.redhat.com/browse/OCPBUGS-86576[OCPBUGS-86576])

Reviewer: ```suggestion
* Before this update, when a user applied a `MachineConfig` parameter to install extensions, the Machine Config Operator (MCO) did not validate that all packages were installed. As a consequence, extension installation appeared to be successful, but packages were missing. If one of more package is not present, the node, and subsequently the associated MachineConfigPool, degraded.  With this release, the MCO validates after the node reboot that all packages associated with your desired extension were successfully installed before reporting a successful update. (link:https://issues.redhat.com/browse/OCPBUGS-86576[OCPBUGS-86576])
```
It's clear this release note was wrong if it had been read.

* Before this update, the macOS Option key was treated as a Meta key instead of a compose key in the pod terminal. As a consequence, characters that rely on Option key combinations, such as `@`, `{`, `}`, `|`, `\`, and `~`, could not be entered. With this release, the terminal correctly identifies macOS, so the Option key functions as a compose key as expected. (link:https://issues.redhat.com/browse/OCPBUGS-86581[OCPBUGS-86581])

> * Before this update, the router liveness probe used an HTTP backend check to  monitor HAProxy health. As a consequence, when HAProxy reached its maxconn connection limit due to client traffic, the HTTP health check failed and Kubernetes unnecessarily restarted the router pods. With this release, the router liveness probe uses an HAProxy admin socket check instead of an HTTP  backend check. As a result, router pods remain stable when HAProxy reaches  its maxconn connection limit due to legitimate client traffic, preventing  unnecessary restarts. (link:https://issues.redhat.com/browse/OCPBUGS-87002[OCPBUGS-87002])

Reviewer: ```suggestion
* Before this update, the router liveness probe used an HTTP backend check to monitor HAProxy health. As a consequence, when HAProxy reached its `maxconn` connection limit due to client traffic, the HTTP health check failed and the router pods unnecessarily restarted. With this release, the router liveness probe uses an HAProxy `admin` socket check instead of an HTTP  backend check. As a result, router pods remain stable when HAProxy reaches  its `maxconn` connection limit due to legitimate client traffic, preventing  unnecessary restarts. (link:https://issues.redhat.com/browse/OCPBUGS-87002[OCPBUGS-87002])
```
personification of Kubernetes (also just not needed)
